editoast/fga_migrations/src/migration.rs

```diff
@@ -0,0 +1,509 @@
+                let tuple_to_remove = Model::apply().tuple(&old_migration, singleton);
+                client_migrations_store
+                    .delete_tuples(&[tuple_to_remove])
+                    .await?;
+                info!("Deleted existing migration tuple");
+            }
+            [_, _, ..] => {
+                error!(
+                    "inconsistent migrations store state: several tuples found for the migration `{}`",
+                    migration_number
+                );
+                panic!()
+            }
+        }
+    }
+    info!("Applying target migration");
+
+    // Build the JSON payload of the HTTP request POST authorization model
+    let new_model = compile_model(migration_model_dsl);
+    info!(openfga.post_stores_authorization_model.payload = new_model.to_string());
+
+    // Add the new model to Editoast OpenFGA store
+    client_authorization_store
+        .update_authorization_model(&new_model)
+        .await?;
+    info!(
+        model.id = client_authorization_store.authorization_model_id,
+        "New model written to authorization store"
+    );
+
+    // Make the tuple of the new migration
+    let migration = Migration(migration_number);
+    let migration_tuple = Model::apply().tuple(&migration, &migration_model);
+
+    // Persist the new DSL model hash in a migration tuple
+    client_migrations_store
+        .write_tuples(&[migration_tuple])
+        .await?;
+    info!("Migration tuple added to authorization store");
+    Ok(client_authorization_store)
+}
+
+#[cfg(any(test, feature = "test_utilities"))]
+pub mod test_utilities {
+    use crate::migration::Settings;
+
+    use super::TelemetrySettings;
+
+    pub fn telemetry_settings() -> TelemetrySettings {
+        let url =
+            std::env::var("TELEMETRY_BACKEND_URL").unwrap_or_else(|_| "localhost".to_string());
+        let port: u16 = std::env::var("TELEMETRY_BACKEND_PORT")
+            .unwrap_or_else(|_| "4317".to_string())
+            .parse()
+            .expect("Invalid telemetry port");
+        TelemetrySettings {
+            endpoint: url::Url::parse(&format!("{}:{}", url, port))
+                .expect("Invalid telemetry settings"),
+        }
+    }
+
+    pub fn settings(authz_store: &str, migrations_store: &str) -> Settings {
+        Settings {
+            migration_settings: super::MigrationSettings {
+                connection_settings: fga::test_utilities::connection_settings(),
+                migration_store: migrations_store.to_string(),
+                authorization_store: authz_store.to_string(),
+            },
+            telemetry: telemetry_settings(),
+        }
+    }
+
+    /// Setup two stores for authorization and migrations. Return a client to each store and a
+    /// compatible [crate::migration::MigrationSettings].
+    ///
+    /// Note: the stores are created empty: they don't have any model yet when returned.
+    #[macro_export]
+    macro_rules! setup_stores {
```

Comment:
I think this could be a function taking two Client in parameters? This way we'd avoid a third level of test client macro nesting.